util.go

```diff
@@ -49,41 +48,6 @@ func suint(n int) uint {
 	return uint(n)
 }
 
-var errNoScheme = errors.New("no scheme")
-var errEmptyURL = errors.New("URL cannot be empty")
-
-func sourceSchemeFromURL(url string) (string, error) {
```

Comment:
Thanks for cleaning up this code! e.g. removing `sourceSchemeFromURL` and  `databaseSchemeFromURL`